feat: Dedicated release note for dropping repository permissions for Administration on GitHub (#1971)
codacy/docs: docs/release-notes/cloud/cloud-2023-12-DD-gh-admin-permission-not-requested.md

---
rss_title: Codacy release notes RSS feed
rss_href: /feed_rss_created.xml
---

> # Codacy no longer requests repository permissions for Administration on GitHub December DD, 2023

Author: Valid notes here, @nicklem! 👍 

Anyway, let's hold any renaming until we align the communication message with CSM.
cc @RaquelHipolito @jorgebraz

> # Codacy no longer requests repository permissions for Administration on GitHub December DD, 2023

Author: @nicklem I renamed the page. WDYT?

!!! info "This release note applies only to GitHub"

For increased security, Codacy is deprecating the usage of **repository SSH keys** for Git operations on GitHub in favour of **installation access tokens**.

> This change translates into an important security improvement by reducing the liability of Codacy GitHub App, as it no longer requests [read and write repository permissions for Administration](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-administration).

Author: Same here. Let's wait until we align the communication message with CSM. 🙏 
cc @RaquelHipolito @jorgebraz

> This change translates into an important security improvement by reducing the liability of Codacy GitHub App, as it no longer requests [read and write repository permissions for Administration](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-administration).

Author: @nicklem I updated this introduction with more information. Please have a look.

@RaquelHipolito @jorgebraz: Please validate the message on this introduction and let me know if we want to add or adjust anything. 🙏

## What do you need to do if you have a GitHub organization? {: id="to-do"}

> Make sure an organization owner [approves Codacy GitHub App updated permissions](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization, if not done yet.

Reviewer: Inverting the sentence here reads better to me 🤔 

```suggestion
Make sure an organization owner [approves the updated Codacy GitHub App permissions](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization, if not done yet.
```

Author: My intent is to match GitHub terminology, [Approving updated permissions for a GitHub App](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app).

WDYT?

Reviewer: How about an alternative that keeps the meaning and still sticks to the GH terminology?

```suggestion
Make sure an organization owner [approves the updated permissions for the Codacy GitHub App](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization, if not done yet.
```

> To use installation access tokens, Codacy GitHub App requires [repository read permissions for Contents](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents), and has requested additional permissions for it.

Reviewer 2: @claudiacarpinteiro "and has requested additional permissions for it." - this scentance got me confused. I am not sure what is this refering to or means 🙈  maybe it's just me tho.

Author: @RaquelHipolito I tweaked the sentence. Let me know if it's clear now. 🙏

> To use installation access tokens, Codacy GitHub App requires [repository read permissions for Contents](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents). Thus, organization owners will be notified to review a request for this additional permission:

Reviewer: It reads better to me with an article, "the". Even better would be "Codacy's GH app", but I guess we chose not to use it. Thoughts?

```suggestion
To use installation access tokens, the Codacy GitHub App requires [repository read permissions for Contents](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents). Thus, organization owners will be notified to review a request for this additional permission:
```

Author: Yes, we have been avoiding that possessive form ('s) because it usually adds complexity to the reading (for technical documentation). I would keep this rule. 👍

## Removal of repository permissions for Administration

To ensure the [Codacy GitHub App updated permissions are approved](#to-do) on your GitHub organization before the removal of repository permissions for Administration, Codacy will execute a phased rollout according to the timeline below:

<table>
  <thead>
    <th>Date</th>
    <th>Event</th>
    <th>What to expect</th>
  </thead>
  <tbody>
    <tr>
      <td>December DD, 2023</td>
      <td>Codacy started using installation access tokens to clone and integrate with your repositories, and deprecated repositories SSH keys usage</td>
      <td>
        <p>From this day on, repositories SSH keys are used exclusively as a fallback mechanism when the Contents permission is missing.</p>
        <p>If not done yet, make sure an organization owner <a href="#to-do">approves Codacy GitHub App updated permissions</a> on your GitHub organization.</p>
      </td>
    </tr>
    <tr>
      <td>January DD, 2024</td>
      <td>Brownout of repository permissions for Administration</td>
      <td>
        <p>On this day, Codacy won't use any fallback mechanism if the Contents permission is missing.</p>

>         <p>This will help you confirm that you already approved the updated permissions for Codacy GitHub App.<br/>If on this date your Codacy quality analysis fail or you have problems adding new repositories to Codacy, make sure an organization owner <a href="#to-do">approves Codacy GitHub App updated permissions</a> on your GitHub organization.</p>

Reviewer: This one is tricky. We should say either `your Codacy quality analysis fails` (singular) or `your Codacy quality analyses fail` (plural), although "analyses" is unusual.

On a side note, another nitpick to add "the":

```suggestion
        <p>This will help you confirm that you already approved the updated permissions for the Codacy GitHub App.<br/>If on this date your Codacy quality analysis fails or you have problems adding new repositories to Codacy, make sure an organization owner <a href="#to-do">approves the updated permissions for the Codacy GitHub App</a> on your GitHub organization.</p>
```

Author: > This one is tricky. We should say either your Codacy quality analysis fails (singular) or your Codacy quality analyses fail (plural), although "analyses" is unusual.

Grammar typo here. 😄 I meant "Codacy quality analysis **fails**" (singular).

Tweaking your version to consider the role that needs to approve the updated permissions. 

 ```suggestion
        <p>This will help you confirm that the updated permissions for the Codacy GitHub App were already approved for your organization.<br/>If on this date your Codacy quality analysis fails or you have problems adding new repositories to Codacy, make sure an organization owner <a href="#to-do">approves the updated permissions for the Codacy GitHub App</a> on your GitHub organization.</p>
```

      </td>
    </tr>
    <tr>
      <td>MM DD, 2024</td>
      <td>Codacy will remove repository permissions for Administration and drop the usage of repository SSH keys</td>
      <td>

>         <p>Codacy will start using exclusively installation access tokens to clone and integrate with your repositories with no fallback mechanisms, and will no longer store repositories SSH keys. Also, repository permissions for Administration will be removed from the Codacy GitHub App.</p>

Reviewer: Tweaks for clarity: 

```suggestion
        <p>Codacy will start using exclusively installation access tokens to clone and integrate with your repositories, with no fallback mechanisms, and will no longer store repository SSH keys. Also, repository permissions for Administration will be removed from the Codacy GitHub App.</p>
```

---
rss_title: Codacy release notes RSS feed
rss_href: /feed_rss_created.xml
---

# Codacy no longer requests repository permissions for Administration on GitHub December DD, 2023

!!! info "This release note applies only to GitHub"

For increased security, Codacy is deprecating the usage of **repository SSH keys** for Git operations on GitHub in favour of **installation access tokens**.

This change translates into an important security improvement by reducing the liability of Codacy GitHub App, as it no longer requests [read and write repository permissions for Administration](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-administration).

## What do you need to do if you have a GitHub organization?

> To use installation access tokens, Codacy GitHub App requires [repository read permissions for Contents](https://docs.github.com/en/enterprise-cloud@latest/rest/overview/permissions-required-for-github-apps?apiVersion=2022-11-28#repository-permissions-for-contents).

Author: I moved the required action to the beginning for better visibility. 👍

Codacy have been requesting updated app permissions for some time now. If you haven't done it yet, make sure you [approve Codacy GitHub App updated permissions](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization.

If you have any questions or need help, please contact <mailto:support@codacy.com>.

## Removal of repository permissions for Administration

> To ensure you have the necessary time to [approve Codacy GitHub App updated permissions](https://docs.github.com/en/apps/using-github-apps/approving-updated-permissions-for-a-github-app) on your GitHub organization, Codacy will keep using repository SSH keys for Git operations for now as a fallback mechanism when the Contents permission is missing.

Author: Yes, makes sense! I removed the details that are already mentioned in the table. 👍
